store/helpers/pages/pages.js

```diff
@@ -0,0 +1,97 @@
+import getOrder from '../order'
+import Page from './page'
+import invalidatePreviousRequests from '../invalidatePreviousRequests'
+
+import { LinkedList } from 'utils/linkedList'
+import apiRequest from 'api'
+import { CancelablePromise } from 'utils/promise'
+
+const PAGE_SIZE = 100
+
+export default class Pages {
+  #pages = new LinkedList()
+
+  #searchTerm = ''
+
+  #columnOrder = ''
+
+  #pageNumber = 0
+
+  isLastPageLoaded = false
+
+  isFirstPageLoaded = false
+
+  constructor(url) {
+    this.url = url
+  }
+
+  async slice(from, to) {
+    while (this.#pages.length < to && !this.isLastPageLoaded)
+      // eslint-disable-next-line no-await-in-loop
+      await this.load().promise
```

Comment:
Interesting. I am afraid to do so. Be careful. Seems to cause unexpected behaviour

Reply:
It's fine to use it in a while loop. It wouldn't work if it is used in forEach callback.